src/cc/frontends/clang/b_frontend_action.cc

```diff
@@ -47,16 +48,35 @@ const char *calling_conv_regs_s390x[] = {"gprs[2]", "gprs[3]", "gprs[4]",
 
 const char *calling_conv_regs_arm64[] = {"regs[0]", "regs[1]", "regs[2]",
                                        "regs[3]", "regs[4]", "regs[5]"};
+
+const char **get_call_conv(void) {
+  const char *archenv = getenv("ARCH");
+  if (!archenv)
+    archenv = getenv("BCC_ARCH");
+
```

Comment:
what is the difference between ARCH and BCC_ARCH? Should we use an environment variable or a cmake macro definition?

Reply:
CMake macro definition would make it static then right? We want it to be dynamic. This is indeed how the kernel is built in a cross compiler environment as well. One does something like this in kernel build directory, for example:
```
ARCH=arm64 make zImage
```
Similarly, we need BCC to adapt to the user's cross compile environment like the kernel does.

Reply:
okay, env seems fine here. What is the different between ARCH and BCC_ARCH?

Reply:
I just thought of giving `BCC_ARCH` as an additional option since most (all?) env vars or macros are prefixed with `BCC_`. There's no difference between them.

Reply:
Can we just one, e.g., `BCC_TARGET_ARCH`? Having two is not really necessary here, in my opinion.

Reply:
Yes I agree we can have one. Hope you agree that lets just keep `ARCH` because this what the kernel uses as well so it will simplify the kernel developer's workflow here.